### ISSUE TITLE
fp feedback should always unblacklist user

### ISSUE THREAD

**Issue by Glorfindel83** [Charcoal-SE/SmokeDetector#1072]
Right now, fp feedback unblacklists a user only when it's given via chat, presumably because SmokeDetector can then directly act on feedback. Other feedback (via FDSC, FIRE, or metasmoke's /review page) is registered in metasmoke first, but should also unblacklist the user.

**Comment by angussidney** [comment]
Now that we use the API almost exclusively, I think this is pretty important to implement. [Similar functions are already implemented](https://github.com/Charcoal-SE/SmokeDetector/blob/master/metasmoke.py#L94), so it shouldn't take too much effort.

**Comment by quartata** [comment]
Forgot to comment on this (although I've discussed this several times in chat). Currently SmokeDetector only unblacklists a user if the report contains the string "Blacklisted user:" (note the capitalization, can only occur if the only reason a report was caught was because the user was blacklisted). This obviously is rather problematic ([what if the title contains "Blacklisted user:"?](https://chat.stackexchange.com/transcript/message/40455420#40455420)), so I've changed it to actually check the blacklist and remove the entry on FP in my chatcommunicate PR. I can change the behavior to be like this "only reason is blacklisted user" if desired however.
